pkg/operations/kubernetesupgrade/upgradecluster.go

```diff
@@ -250,16 +206,19 @@ func (uc *UpgradeCluster) getClusterNodeStatus(subscriptionID uuid.UUID, az armh
 	return nil
 }
 
-// getClusterNodeVersion returns a node's "orchestrator:version" via Kubernetes API or VM tag.
-func (uc *UpgradeCluster) getClusterNodeVersion(client armhelpers.KubernetesClient, name string, tags map[string]*string) string {
+// getNodeVersion returns a node's current Kubernetes version via Kubernetes API or VM tag.
+func (uc *UpgradeCluster) getNodeVersion(client armhelpers.KubernetesClient, name string, tags map[string]*string) string {
 	if tags != nil && tags["orchestrator"] != nil {
-		return *tags["orchestrator"]
+		parts := strings.Split(*tags["orchestrator"], ":")
+		if len(parts) == 2 {
+			return parts[1]
+		}
 	}
-	uc.Logger.Warnf("Expected tag \"orchestrator\" not found for VM: %s", name)
+	uc.Logger.Warnf("Expected tag \"orchestrator\" not found for VM: %s. Using Kubernetes API to retrieve Kubernetes version.", name)
 	if client != nil {
 		node, err := client.GetNode(name)
 		if err == nil {
-			return api.Kubernetes + ":" + strings.TrimPrefix(node.Status.NodeInfo.KubeletVersion, "v")
```

Comment:
here we return just the k8s version instead of `kubernetes:v1.x.x` since this is in the `kubernetesupgrade` package, we only care about the version - not the orchestrator.